test/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/EnterOrChangeContactDetailsControllerSpec.scala

```diff
@@ -0,0 +1,269 @@
+import play.api.test.FakeRequest
+import play.api.test.Helpers.{BAD_REQUEST, contentAsString, defaultAwaitTimeout}
+import uk.gov.hmrc.auth.core.AuthConnector
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.cache.SessionCache
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.claims.EnterOrChangeContactDetailsController.mrnContactDetailsForm
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.{AuthSupport, ControllerSpec, SessionSupport, routes => baseRoutes}
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.DraftClaim.DraftC285Claim
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.JourneyStatus.FillingOutClaim
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models._
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.email.Email
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.generators.EmailGen._
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.generators.PhoneNumberGen._
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.phonenumber.PhoneNumber
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.generators.MrnContactDetailsGen._
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.generators.SignedInUserDetailsGen._
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.generators.Generators.{sample, _}
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.generators.IdGen._
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.ids.GGCredId
+
+import scala.concurrent.Future
+
+class EnterOrChangeContactDetailsControllerSpec
+    extends ControllerSpec
+    with AuthSupport
+    with SessionSupport
+    with TableDrivenPropertyChecks {
+
+  override val overrideBindings: List[GuiceableModule] =
+    List[GuiceableModule](
+      bind[AuthConnector].toInstance(mockAuthConnector),
+      bind[SessionCache].toInstance(mockSessionCache)
+    )
+
+  lazy val controller: EnterOrChangeContactDetailsController = instanceOf[EnterOrChangeContactDetailsController]
+
+  implicit lazy val messagesApi: MessagesApi = controller.messagesApi
+
+  implicit lazy val messages: Messages = MessagesImpl(Lang("en"), messagesApi)
+
+  val signedInUserEmail = sample[Email]
+
+  private val journeys = Table(
+    "JourneyBindable",
+    JourneyBindable.Single,
+    JourneyBindable.Bulk,
+    JourneyBindable.Scheduled
+  )
+
+  private def getSessionWithPreviousAnswer(
+    maybeMrnContactDetailsAnswer: Option[MrnContactDetails]
+  ): (SessionData, FillingOutClaim) = {
+    val draftC285Claim      = DraftC285Claim.newDraftC285Claim.copy(mrnContactDetailsAnswer = maybeMrnContactDetailsAnswer)
+    val ggCredId            = sample[GGCredId]
+    val signedInUserDetails = sample[SignedInUserDetails]
+    val journey             = FillingOutClaim(ggCredId, signedInUserDetails, draftC285Claim)
+    (SessionData.empty.copy(journeyStatus = Some(journey)), journey)
+  }
+
+  def performAction(mrnContactDetails: MrnContactDetails)(action: Action[AnyContent]): Future[Result] =
```

Comment:
This looks like overcomplicating it.
One method calling another, calling the action ...